core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -55,47 +54,69 @@
  * distributed with Alluxio jar. Alluxio users can override values of these default properties by
  * creating {@code alluxio-site.properties} and putting it under java {@code CLASSPATH} when running
  * Alluxio (e.g., ${ALLUXIO_HOME}/conf/)
- *
- * <p>
- * Developers can create an instance of this class by {@link #Configuration()}, which will load
- * values from any Java system properties set as well.
- *
- * <p>
- * The class only supports creation using {@link #Configuration(Properties)} to override default
- * values.
  */
 @NotThreadSafe
 public final class Configuration {
   /** File to set default properties. */
   public static final String DEFAULT_PROPERTIES = "alluxio-default.properties";
-  /** File to set customized properties. */
+  /** File to set customized properties for Alluxio deployment. */
   public static final String SITE_PROPERTIES = "alluxio-site.properties";
-  /** Regex string to find ${key} for variable substitution. */
+  /** File to set customized properties for Alluxio server. */
+  public static final String SERVER_PROPERTIES = "alluxio-server.properties";
+  /** File to set customized properties for Alluxio client. */
+  public static final String CLIENT_PROPERTIES = "alluxio-client.properties";
+
+  /** Regex string to find "${key}" for variable substitution. */
   private static final String REGEX_STRING = "(\\$\\{([^{}]*)\\})";
   /** Regex to find ${key} for variable substitution. */
   private static final Pattern CONF_REGEX = Pattern.compile(REGEX_STRING);
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
-
+  /** Set of properties. */
   private final Properties mProperties = new Properties();
 
   /**
-   * Overrides default properties.
-   *
-   * @param props override {@link Properties}
+   * @param conf the source configuration to copy
+   * @return a new configuration instance with the properties copied from source configuration
    */
-  public Configuration(Map<String, String> props) {
-    if (props != null) {
-      mProperties.putAll(props);
-    }
-    checkUserFileBufferBytes();
+  public static Configuration copyFrom(Configuration conf) {
+    return new Configuration(conf.toMap());
+  }
+
+  /**
+   * @param properties the source configuration
+   * @return a new configuration with only properties copied from source
+   */
+  public static Configuration fromMap(Map<?, ?> properties) {
+    return new Configuration(properties);
+  }
+
+  /**
+   * @return the default configuration without loading site and system properties
+   */
+  public static Configuration createDefaultConf() {
```

Comment:
that's in the followup PR (adding unit tests for those constructors and utils)
TODO added